### PR TITLE
Refactor #10078 v98 XCUITests update FullFunctional Test suite

### LINKED ISSUE #10078: [XCUITests] Full functional tests need to be updated after new browser tab menu lands
There are a lot of changes with the removal of the POM and some of those options moved into BrowserTabMenu, other options have been deleted and there are a few new ones.
See PR #10072 

Tests for the Smoketest are updated in the same PR that lands with those changes so that Bitrise is green, but the rest of tests should be updated too.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-3804)

### PR DESCRIPTION
**Pull request by isabelrios**
Fixes #10078

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -150,7 +150,8 @@ workflows:
         inputs:
         - scheme: Fennec_Enterprise_XCUITests
         - xcodebuild_test_options: "-testPlan Fennec_Enterprise_XCUITests"
-        - simulator_device: "$IOS_DEVICE"
+        #- simulator_device: "$IOS_DEVICE"
+        - simulator_device: iPhone 11
```
Comment (isabelrios) [inline bitrise.yml:154]: TODO: Change back to previous value

**Comment by isabelrios** [comment]
@AaronMT @SimonBasca could I please get your review on this? we would need the test fixes for L10n screenshots as a new round is required soon. Thank you 🙏

File: L10nSnapshotTests/L10nSuite2SnapshotTests.swift [hunk 2]
```diff
@@ -115,6 +105,7 @@ class L10nSuite2SnapshotTests: L10nBaseSnapshotTests {
         app.buttons["urlBar-cancel"].tap()
         navigator.nowAt(NewTabScreen)
         navigator.goto(SettingsScreen)
+        print(app.debugDescription)
```
Comment (SimonBasca) [inline L10nSnapshotTests/L10nSuite2SnapshotTests.swift:108]: Nit: Remove debugDescription.

File: XCUITests/ReaderViewUITest.swift [hunk 3]
```diff
@@ -149,9 +149,10 @@ class ReaderViewTest: BaseTestCase {
         addContentToReaderView()
         navigator.goto(BrowserTabMenu)
         navigator.goto(LibraryPanel_ReadingList)
-        navigator.goto(LibraryPanel_ReadingList)
+//        navigator.goto(LibraryPanel_ReadingList)
```
Comment (SimonBasca) [inline XCUITests/ReaderViewUITest.swift:152]: Nit: I think this can be removed.

File: XCUITests/SearchTest.swift [hunk 4]
```diff
@@ -248,14 +248,14 @@ class SearchTests: BaseTestCase {
...
         }
         
         app.buttons[AccessibilityIdentifiers.Toolbar.homeButton].tap()
 
-        waitForExistence(app.buttons["urlBar-cancel"])
-        app.buttons["urlBar-cancel"].tap()
-        print(app.debugDescription)
+//        waitForExistence(app.buttons["urlBar-cancel"])
+//        app.buttons["urlBar-cancel"].tap()
+//        print(app.debugDescription)
```
Comment (SimonBasca) [inline XCUITests/SearchTest.swift:258]: Remove comments and "debugDescription"

File: XCUITests/ReaderViewUITest.swift [hunk 5]
```diff
@@ -163,20 +164,21 @@ class ReaderViewTest: BaseTestCase {
...
+    func testAddToReadingListFromBrowserTabMenu() {
         navigator.performAction(Action.CloseURLBarOpen)
         navigator.nowAt(NewTabScreen)
         // First time Reading list is empty
         navigator.goto(LibraryPanel_ReadingList)
         checkReadingListNumberOfItems(items: 0)
         app.buttons["Done"].tap()
         // Add item to Reading List from Page Options Menu
-        userState.url = path(forTestPage: "test-mozilla-book.html")
+//        userState.url = path(forTestPage: "test-mozilla-book.html")
```
Comment (SimonBasca) [inline XCUITests/ReaderViewUITest.swift:175]: Remove comments.

File: XCUITests/ReaderViewUITest.swift [hunk 6]
```diff
@@ -163,20 +164,21 @@ class ReaderViewTest: BaseTestCase {
...
         updateScreenGraph()
-        navigator.goto(BrowserTab)
+//        navigator.goto(BrowserTab)
```
Comment (SimonBasca) [inline XCUITests/ReaderViewUITest.swift:177]: Remove comments.

File: XCUITests/ReaderViewUITest.swift [hunk 7]
```diff
@@ -163,20 +164,21 @@ class ReaderViewTest: BaseTestCase {
...
+        navigator.openURL(path(forTestPage: "test-mozilla-org.html"))
         waitUntilPageLoad()
-        navigator.browserPerformAction(.addReadingListOption)
-
+//        navigator.browserPerformAction(.addReadingListOption)
```
Comment (SimonBasca) [inline XCUITests/ReaderViewUITest.swift:180]: Remove comments.

**Review by SimonBasca — CHANGES_REQUESTED** [review]
Looks good, just a few comments need addressing.

File: XCUITests/ReaderViewUITest.swift [hunk 8]
```diff
@@ -192,12 +190,14 @@ class ReaderViewTest: BaseTestCase {
         navigator.goto(LibraryPanel_ReadingList)
 
         // Long tap on the item just saved
+        print(app.debugDescription)
```
Comment (SimonBasca) [inline XCUITests/ReaderViewUITest.swift:193]: Nit: Remove `print(app.debugDescription)`

**Review by SimonBasca — APPROVED** [review]

**Pull request merged**